gef-ui/server/controllers/amendments/do-you-have-a-facility-end-date/post-do-you-have-a-facility-end-date.ts

```diff
@@ -0,0 +1,56 @@
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { Response } from 'express';
+import * as api from '../../../services/api';
+import { DoYouHaveAFacilityEndDateViewModel } from '../../../types/view-models/amendments/do-you-have-a-facility-end-date-view-model';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { getNextPage } from '../helpers/navigation.helper';
+import { PORTAL_AMENDMENT_PAGES } from '../../../constants/amendments';
+import { validateIsUsingFacilityEndDate } from './validation';
+import { validationErrorHandler } from '../../../utils/helpers';
+
+export type PostDoYouHaveAFacilityEndDateRequest = CustomExpressRequest<{
+  params: { dealId: string; facilityId: string; amendmentId: string };
+  reqBody: { isUsingFacilityEndDate: string | undefined; previousPage: string };
+}>;
+
+/**
+ * Controller to update the isUsingFacilityEndDate value
+ * @param req - the request object
+ * @param res - the response object
+ */
+export const postDoYouHaveAFacilityEndDate = async (req: PostDoYouHaveAFacilityEndDateRequest, res: Response) => {
+  try {
+    const { dealId, facilityId, amendmentId } = req.params;
+    const { isUsingFacilityEndDate, previousPage } = req.body;
+    const { userToken } = asLoggedInUserSession(req.session);
+
+    const deal = await api.getApplication({ dealId, userToken });
+    const { details: facility } = await api.getFacility({ facilityId, userToken });
+
+    if (!deal || !facility) {
+      console.error('Deal %s or Facility %s was not found', dealId, facilityId);
+      return res.redirect('/not-found');
+    }
+
+    const errorsOrValue = validateIsUsingFacilityEndDate(isUsingFacilityEndDate);
+
+    if ('errors' in errorsOrValue) {
+      const viewModel: DoYouHaveAFacilityEndDateViewModel = {
+        exporterName: deal.exporter.companyName,
+        cancelUrl: `/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/cancel`,
+        previousPage,
+        errors: validationErrorHandler(errorsOrValue.errors),
+        isUsingFacilityEndDate,
+      };
+
+      return res.render('partials/amendments/do-you-have-a-facility-end-date.njk', viewModel);
+    }
+
+    const amendment = await api.updateAmendment({ facilityId, amendmentId, update: { isUsingFacilityEndDate: errorsOrValue.value }, userToken });
```

Comment:
Can `update` be declared above as a const and passed just as an argument, to maintain code conistency please.